### ISSUE TITLE
[Bug]  Cannot activate the "Bambu Cloud Configuration

### ISSUE THREAD

**Issue by mash55** [greghesp/ha-bambulab#682]
### Describe the bug

Failed to connect. Check credentials. - when going through the initial (1st time) setup.

![Bambu-1](https://github.com/user-attachments/assets/07951f6b-1692-4df3-9795-9d8723140c74)

### To Reproduce

- Version:  v2.0.28-code3
- Located in the US
- Changed PW in my Bambu account as described in the docu
- Used the credentials in a web browser successfully
- Same credentials in 'bambulab cloud integration' setup generates the error

### Expected Behaviour

Being able to LOGIN to my Bambulab account

### What device are you using?

A1

### Diagnostic Output

```shell
see attachment 
![Bambu-1](https://github.com/user-attachments/assets/07951f6b-1692-4df3-9795-9d8723140c74)
```

### Other Information

Tried BRAVE & Chrome browsers with same result [probably not related]

**Comment by owenvoke** [comment]
I get the same issue on v2.0.28, with the following debug logs:

```txt
2024-11-05 22:33:01.284 DEBUG (SyncWorker_60) [custom_components.bambu_lab.pybambu] login()
2024-11-05 22:33:01.284 DEBUG (SyncWorker_60) [custom_components.bambu_lab.pybambu] Getting accessToken from Bambu Cloud
2024-11-05 22:33:01.686 DEBUG (SyncWorker_60) [custom_components.bambu_lab.pybambu] Requested loginType: tfa
2024-11-05 22:33:01.688 DEBUG (MainThread) [custom_components.bambu_lab] RESULT = tfa
```

I have 2FA enabled on my account, but assumed it should be ok.

**Comment by mash55** [comment]
I just saw the new release v2.0.28 [without the -code3]. Its now working for me.
Thanks

**Comment by AdrianGarside** [comment]
"Should this line be checking for result == 'tfa' instead?" - yes, I'll get a fix out.

**Comment by AdrianGarside** [comment]
@owenvoke should be fixed in https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.29. I still haven't tested the 2FA flow myself so please let me know if it's now working for you.

**Comment by owenvoke** [comment]
@AdrianGarside, thanks for the fix (it seems to have resolved that issue). Looks like the 2FA part gets blocked by Cloudflare, so I'll just skip using it for now.

**Comment by Y4nnickB** [comment]
> @owenvoke should be fixed in https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.29. I still haven't tested the 2FA flow myself so please let me know if it's now working for you.

I just installed 2.0.29 but seems like it's not fixed. I still get the "Check credentials" warning.

**Comment by AdrianGarside** [comment]
What do debug logs show?

**Comment by Y4nnickB** [comment]
This is the only thing I can't find in the logs:

2024-11-07 08:58:53.218 ERROR (SyncWorker_14) [custom_components.bambu_lab.pybambu] Login attempt failed with error code: 403
2024-11-07 08:58:53.219 ERROR (MainThread) [custom_components.bambu_lab] Failed to connect with error code (403,)

**Comment by AdrianGarside** [comment]
Make sure you're running v2.0.31 to get the latest cloudflare fixes.

Debug logs capturing the initial integration add attempts need to be enabled via a configuration.yaml edit (plus HA restart):
```
logger:
  default: info
  logs:
    custom_components.bambu_lab: debug
```

**Comment by Y4nnickB** [comment]
Hi Adrian,

I just installed version 2.0.31 and after 2 tries I was able to add a printer. I tried adding my 2nd printer but I keep getting the error.
The debug logging states I'm blocked from bambulab.com. I suspect their Cloudflare protection?
[home-assistant_bambu_lab_2024-11-07T08-09-57.101Z.log](https://github.com/user-attachments/files/17658190/home-assistant_bambu_lab_2024-11-07T08-09-57.101Z.log)
